### ISSUE TITLE
REST-API for game

### ISSUE THREAD

**Issue by Laserlicht** [vcmi/vcmi#4702]
**Describe your proposal**
Not high priority, but want to save it as idea:

Provide API to access data from VCMI game over REST-API. For example for accessing the GameStatistics.

Would make it possible to show ingame variables e.g. with OBS in live streams ([see here](https://obsproject.com/forum/resources/url-api-source-live-data-media-and-ai-on-obs-made-simple.1756/)). E.g. income, number of mines, etc.

**Comment by gestern** [comment]
Oh, this is a sure way to lead a project to multiple vulnerabilities. Though any analogies are false, but as soon as a desktop product has such a feature, it's a reason for users to **ban any requests from the game** to open ports.

**Comment by Laserlicht** [comment]
You are aware that VCMI already open a network port for playing? So not much things are changing...
And it would also be possible to bind only on localhost, that this port is only accessible on same host.

**Comment by gestern** [comment]
Yes I'm aware of that because my firewall reminds me of it every time. And yes, I don't consider it safe. I don't see why I should open ports for external communication when I just want to play a game with my PC.

**Comment by Laserlicht** [comment]
`external communication`

As said: Bind to localhost shouldn't be more unsafe than other IPC variants (and should btw not open firewall remainder as it cannot accessed over network). The advantage of REST over other IPC variants would be the interoperability (works independent from OS, programming language, etc.; we can use existing plugins)
